gmqtt/message_queue.py

```diff
@@ -0,0 +1,125 @@
+import asyncio
+import enum
+import functools
+import collections
+
+# from .aioclient import Subscription
+from .client import Message
+
+from typing import List, Tuple, Dict, Callable
+
+
+class TopicFilter:
+    def __init__(self, topic_filter: str):
+        self.levels = topic_filter.split("/")
+        TopicFilter.validate_topic(self.levels)
```

Comment:
Do you have some reason to call `validate_topic` by naming `TopicFilter`? 

It will generate some unexpected behavior if you will inherit from it (if you will inherit and will want to overwrite validate_topic, also you will need to overwrite `__init__` and call `super` and call `validate_topic` manually):
```
class B(TopicFilter):
    @staticmethod
    def validate_topic(*args):
        print("Never happens")
        return super().validate_topic(*args)

B("my-awesome-topic/bla")
```